PixelHunter1995/InventoryLib/Inventory.cs

```diff
@@ -61,12 +59,35 @@ public void Draw(GraphicsDeviceManager graphics, SpriteBatch spriteBatch, double
                 Add("banana", "dogs");
                 Add("apple", "dogs");
                 Add("broccoli", "dogs");
+                Add("broccoli", "dogs");
+                Add("broccoli", "dogs");
+                Add("broccoli", "dogs");
+                Add("broccoli", "dogs");
+                Add("broccoli", "dogs");
                 System.Console.WriteLine(this);
             }
-            spriteBatch.Draw(InventoryBackground, new Vector2(X_POS, Y_POS), Color.White);
-            for (int i = 0; i < Items.Count; i++)
+            DrawBackground(graphics, spriteBatch);
+            foreach (InventoryItem item in Items)
+            {
+                item.Draw(graphics, spriteBatch, scaling);
+            }
+        }
+
+        private void DrawBackground(GraphicsDeviceManager graphics, SpriteBatch spriteBatch)
+        {
+            // Draw background boxes for each visible item slot
+            Texture2D rect = new Texture2D(graphics.GraphicsDevice, ITEM_WIDTH, ITEM_HEIGHT);
+            Color[] data = new Color[ITEM_WIDTH * ITEM_HEIGHT];
+            for (int i = 0; i < data.Length; ++i) data[i] = Color.Gray;
```

Comment:
I would prefer brackets even in one-liners, though it is purely because I subjectively find it more readable.
`for (int i = 0; i < data.Length; ++i) { data[i] = Color.Gray; }`